media.go

```diff
@@ -128,67 +169,30 @@ func (wac *Conn) Upload(reader io.Reader, appInfo MediaType) (url string, mediaK
 	sha.Write(append(enc, mac...))
 	fileEncSha256 = sha.Sum(nil)
 
-	var filetype string
-	switch appInfo {
-	case MediaImage:
-		filetype = "image"
-	case MediaAudio:
-		filetype = "audio"
-	case MediaDocument:
-		filetype = "document"
-	case MediaVideo:
-		filetype = "video"
-	}
-
-	uploadReq := []interface{}{"action", "encr_upload", filetype, base64.StdEncoding.EncodeToString(fileEncSha256)}
-	ch, err := wac.writeJson(uploadReq)
-	if err != nil {
-		return "", nil, nil, nil, 0, err
+	hostname, auth, _, err := wac.queryMediaConn()
```

Comment:
don't we want to cache the token by the TTL?

Reply:
@tulir what do you think?

Reply:
Works fine without caching too, but it can be added later if someone else wants to do it